### ISSUE TITLE
compiling error

### ISSUE THREAD

**Issue by jianshu93** [baku4/lt-fm-index#2]
Hello,

This error is alway on macOS ARM64 platforms:

error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> src/structure/construction/nc_with_noise.rs:205:12
    |
205 | pub struct BwtBlock128NN {
    |            ^^^^^^^^^^^^^
    |
    = note: source type: `BwtBlock128NN` (768 bits)
    = note: target type: `nc_with_noise::_::{closure#0}::TypeWithoutPadding` (704 bits)

error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> src/structure/construction/aa_with_noise.rs:596:12
    |
596 | pub struct BwtBlock128AN {
    |            ^^^^^^^^^^^^^
    |
    = note: source type: `BwtBlock128AN` (2048 bits)
    = note: target type: `aa_with_noise::_::{closure#0}::TypeWithoutPadding` (1984 bits)

For more information about this error, try `rustc --explain E0512`.
error: could not compile `lt-fm-index` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...

Thanks,

Jianshu

**Comment by baku4** [comment]
Thank you for the bug report!

This problem seems to be caused by deriving `Pod` trait in the `bytemuck` crate.

I added custom padding to struct and updated the fixed version to **0.5.6**.

Let me know if there is still a problem.

**Comment by jianshu93** [comment]
Hello,

It works now on ARM64.

Thanks,

Jianshu

**Comment by marade** [comment]
This bug appears to have resurfaced in 0.6.0 on x86_64:

```
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lt-fm-index-0.6.0/src/structures/construction/nc_with_noise.rs:205:12
    |
205 | pub struct BwtBlock128NN {
    |            ^^^^^^^^^^^^^
    |
    = note: source type: `BwtBlock128NN` (768 bits)
    = note: target type: `nc_with_noise::_::{closure#0}::TypeWithoutPadding` (704 bits)

error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lt-fm-index-0.6.0/src/structures/construction/aa_with_noise.rs:596:12
    |
596 | pub struct BwtBlock128AN {
    |            ^^^^^^^^^^^^^
    |
    = note: source type: `BwtBlock128AN` (2048 bits)
    = note: target type: `aa_with_noise::_::{closure#0}::TypeWithoutPadding` (1984 bits)

For more information about this error, try `rustc --explain E0512`.
error: could not compile `lt-fm-index` (lib) due to 2 previous errors
```

**Comment by baku4** [comment]
@marade 

Thank you for reporting the issue.
It has been fixed in version 0.6.1 (https://crates.io/crates/lt-fm-index/0.6.1).
Have a great day.

**Comment by marade** [comment]
@baku4 Thank you!
